src/lib/components/Swap/Output.tsx

```diff
@@ -68,7 +74,14 @@ export default function Output({ disabled, children }: OutputProps) {
             <Trans>For</Trans>
           </ThemedText.Subhead2>
         </Row>
-        <TokenInput input={output} disabled={disabled} onChangeInput={setValue} onChangeToken={setToken}>
+        <TokenInput
+          currency={outputCurrency}
+          amount={(typedOutputAmount !== undefined ? typedOutputAmount : outputCurrencyAmount?.toSignificant(6)) ?? ''}
+          disabled={disabled}
+          onMax={balance ? () => updateTypedOutputAmount(balance.toExact()) : undefined}
```

Comment:
use a `useCallback` for `onMax`, and pass the setters directly to `onChangeInput`/`onChangeCurrency`.